src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c

```diff
@@ -21,26 +29,250 @@ const ActorInit Obj_Bell_InitVars = {
+    Vec3f bumperPos;
+    Vec3f worldPos;
+    s32 phi_a3 = false;
+
+    if (((arg1 == 0) && (this->unk_21C < 1000.0f)) || ((arg1 == 1) && (this->unk_21C < 4000.0f)) || (arg1 == 2)) {
+        phi_a3 = true;
+    } else {
+        phi_a3 = phi_a3;
+    }
+
+    switch (arg1) {
+        case 0:
+            this->unk_21C += this->unk_21C > 1000.0f ? 250.0f : 1000.0f;
+            break;
+        case 1:
+            this->unk_21C += this->unk_21C > 3000.0f ? 750.0f : 3000.0f;
+            break;
+        case 2:
+            if (1) {}
+            this->unk_21C += 9000.0f;
+            break;
+    }
+
+    this->unk_21C = CLAMP(this->unk_21C, 0.0f, 18000.0f);
+    if (phi_a3 == true) {
+        Math_Vec3s_ToVec3f(&bumperPos, &this->collider2.info.bumper.hitPos);
+        Math_Vec3f_Copy(&worldPos, &this->dyna.actor.world.pos);
+        this->dyna.actor.world.rot.y = Math_Vec3f_Yaw(&bumperPos, &worldPos);
+        if (this->unk_20C <= 0x4000 && this->unk_20C >= -0x4000) {
+            this->unk_20C -= 0x4000;
+        } else {
+            this->unk_20C += 0x4000;
+        }
+    }
+    return phi_a3;
+}
+
+s32 func_80A356D8(ObjBell* this) {
+    f32 scaleProjection;
+
+    this->unk_212 = this->dyna.actor.world.rot.y;
+    if (this->unk_20C >= 0x4000 || this->unk_20C <= -0x4000) {
+        this->unk_212 -= 0x8000;
+    }
+    scaleProjection = Math_SinS(this->unk_20C) * this->unk_21C;
+    this->dyna.actor.world.rot.x = this->dyna.actor.home.rot.x;
+    this->unk_220 = scaleProjection;
+    this->dyna.actor.world.rot.x += (s16)scaleProjection;
+    Math_ApproachF(&this->unk_21C, 0.0f, 0.03f, 70.0f);
+    if (this->unk_21C > 0.0f) {
+        this->unk_20C -= 0x800;
+    }
+    return false;
+}
+
+s32 func_80A357A8(ObjBell* this, GlobalContext* globalCtx) {
+    f32 temp_f0;
+    s16 temp_v1;
+
+    if (this->collider1.base.ocFlags1 & 2) {
+        this->collider1.base.ocFlags1 &= ~2;
+        temp_v1 = this->dyna.actor.yawTowardsPlayer - this->unk_212;
+        temp_f0 = this->unk_21C / 18000.0f;
+        if (ABS_ALT(temp_v1) < 0x3FFC) {
+            if (this->unk_214 == 0) {
+                if (temp_f0 > 0.18f) {
+                    func_800B8D98(globalCtx, &this->dyna.actor, 8.0f * temp_f0, this->dyna.actor.yawTowardsPlayer,
+                                  11.0f * temp_f0);
+                    this->unk_214 = 30;
+                }
+            }
+        }
+    }
+    if (this->collider2.base.acFlags & 2) {
+        this->collider2.base.acFlags &= ~2;
+        this->unk_20E = 10;
+        switch (this->dyna.actor.colChkInfo.damageEffect) {
+            case 15:
+                Audio_PlayActorSound2(this, 0x28F3);
+                func_80A35510(this, 1);
+                break;
+            case 14:
+                Audio_PlayActorSound2(this, 0x28F3);
+                func_80A35510(this, 2);
+                break;
+            default:
+                func_80A35510(this, 0);
+                break;
+        }
+    }
+    return false;
```

Comment:
Thanks for the fixes :) one last thought. Three functions, `func_80A35510`, `func_80A356D8`, and `func_80A357A8`, all have return values that seem unnecessary, as the return is never used when any of these functions are called. Sometimes a return is required to match, but is there any chance you can remove the `return false` switch to `void func_` for any of them and still get a match?

Reply:
![image](https://user-images.githubusercontent.com/20159000/120249664-97cb0d00-c249-11eb-8626-d8328f9acabb.png)